tools/sourcemap-tools/src/ZipArchive.ts

```diff
@@ -0,0 +1,35 @@
+import archiver from 'archiver';
+import { Readable, Transform, TransformCallback, TransformOptions } from 'stream';
+
+export class ZipArchive extends Transform {
+    private readonly _archive: archiver.Archiver;
+
+    constructor(opts?: TransformOptions) {
+        super(opts);
+        this._archive = archiver('zip');
+    }
+
+    public append(name: string, sourceMap: string | Readable | Buffer) {
+        this._archive.append(sourceMap, { name });
+        return this;
+    }
+
+    public finalize() {
+        return this._archive.finalize();
+    }
+
+    public override pipe<T extends NodeJS.WritableStream>(
+        destination: T,
+        options?: { end?: boolean | undefined } | undefined,
```

Comment:
Since `end` is already optional, do you need the `| undefined`?

Reply:
We do not, that was automatically inserted by VS Code. Will fix